StyleCop.Analyzers/StyleCop.Analyzers.Test/NamingRules/SA1302UnitTests.cs

```diff
@@ -185,6 +185,160 @@ public interface FileOpenDialog111
             await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
         }
 
+        [Fact]
+        public async Task TestInterfaceDeclarationDoesNotStartWithIWithConflictAsync()
+        {
+            string testCode = @"
+public interface Foo
+{
+}
+
+public interface IFoo { }";
+            string fixedCode = @"
+public interface IFoo1
+{
+}
+
+public interface IFoo { }";
+
+            DiagnosticResult expected = this.CSharpDiagnostic().WithLocation(2, 18);
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpDiagnosticAsync(fixedCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpFixAsync(testCode, fixedCode, cancellationToken: CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestInterfaceDeclarationDoesNotStartWithIWithMemberConflictAsync()
+        {
+            string testCode = @"
+public interface Foo
+{
+    int IFoo { get; }
+}";
+            string fixedCode = @"
+public interface IFoo1
+{
+    int IFoo { get; }
+}";
+
+            DiagnosticResult expected = this.CSharpDiagnostic().WithLocation(2, 18);
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpDiagnosticAsync(fixedCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+            await this.VerifyCSharpFixAsync(testCode, fixedCode, cancellationToken: CancellationToken.None).ConfigureAwait(false);
+        }
```

Comment:
:arrow_right: It contains a member named `IFoo`. Members are not allowed to have the same name as the containing type, so either we don't use the name `IFoo` for the interface, or we have to rename both the interface and the property.